javalib/src/main/scala/java/net/URI.scala

```diff
@@ -41,18 +41,19 @@ final class URI(origStr: String) extends Serializable with Comparable[URI] {
     else fld(AbsHierPart)
   }.get
 
-  private val _authority = fld(AbsAuthority, RelAuthority)
+  private val _authority = fld(AbsAuthority, RelAuthority).filter(_ != "")
   private val _userInfo = fld(AbsUserInfo, RelUserInfo)
   private val _host = fld(AbsHost, RelHost)
   private val _port = fld(AbsPort, RelPort).fold(-1)(_.toInt)
 
   private val _path = {
-    if (_isAbsolute) {
-      if (_authority.isDefined) fld(AbsNetPath)
-      else fld(AbsAbsPath)
+    if (fld(AbsAuthority, RelAuthority).isDefined) {
```

Comment:
Yeah. You are right. So make it a local val. But like this it is very hard to understand what your intent is.